packages/core/src/sdk/session/index.ts

```diff
@@ -50,6 +50,16 @@ export const mutation = gql(`
 `)
 
 export const validateSession = async (session: Session) => {
+  if (storeConfig.deliveryPromise?.enabled && !session.postalCode) {
+    // Use the initial postalCode defined in discovery.config.js when there is no postalCode in the session
+    const initialPostalCode = defaultStore.readInitial().postalCode
+
+    if (!!initialPostalCode) {
+      session = { ...session, postalCode: initialPostalCode }
+      sessionStore.set(session)
```

Comment:
just for imutability sake
```suggestion
      sessionStore.set({ ...session, postalCode: initialPostalCode })
```